parameters.txt
//******************************
//  Simulation Parameters
//******************************

rnd_seed = 5													// random number for KMCProcess
restart = 0														// restart calculation
restart_xyz_file = ./final_parameters/snapshot_109_setsweep.xyz	// restart filename

atom_xyz_file = structures/old_structures/atoms.xyz				// if not restart, draws from the atom coordinates file
interstitial_xyz_file = structures/old_structures/interstitials.xyz
lattice = 93.72586 26.914773122 26.6371955996 					// 1x1 device

pristine = 1													// add initial vacancy concentration if starting from pristine
initial_vacancy_concentration = 0.05 
attempt_frequency = 10e13 										// [1/s]

shift = 0														// translate cell across the periodic boundary
shifts = 0.0 0.5 0.5

nn_dist = 3.5 													// [Angstrom] neighbor distance

perturb_structure = 1											// KMC event step ( turn off to only calculate fields )
solve_potential = 1												// field solvers
solve_current = 1
solve_heating_global = 1
solve_heating_local = 0
																// for potential solver:
sigma = 3.5e-10 												// [m] gaussian broadening
epsilon = 23.0  												// [1] relative permittivity
								
																// for current solver (tunneling parameters)
m_r = 0.85														// [1] relative effective mass
V0 = 1.6  														// [eV] defect state energy

																// for temperature solver
k_therm = 1.1         											// [W/mK] thermal conductivity
background_temp = 300 											// [K] contact temperature

pbc = 1															// toggle for periodic boundary conditions

V_switch = 5 //0 0.0404040404040404 0.0808080808080808 0.121212121212121 0.161616161616162 0.202020202020202 0.242424242424242 0.282828282828283 0.323232323232323 0.363636363636364 0.404040404040404 0.444444444444444 0.484848484848485 0.525252525252525 0.565656565656566 0.606060606060606 0.646464646464647 0.686868686868687 0.727272727272727 0.767676767676768 0.808080808080808 0.848484848484849 0.888888888888889 0.929292929292929 0.96969696969697 1.01010101010101 1.05050505050505 1.09090909090909 1.13131313131313 1.17171717171717 1.21212121212121 1.25252525252525 1.29292929292929 1.33333333333333 1.37373737373737 1.41414141414141 1.45454545454545 1.4949494949495 1.53535353535354 1.57575757575758 1.61616161616162 1.65656565656566 1.6969696969697 1.73737373737374 1.77777777777778 1.81818181818182 1.85858585858586 1.8989898989899 1.93939393939394 1.97979797979798 2.02020202020202 2.06060606060606 2.1010101010101 2.14141414141414 2.18181818181818 2.22222222222222 2.26262626262626 2.3030303030303 2.34343434343434 2.38383838383838 2.42424242424242 2.46464646464646 2.50505050505051 2.54545454545455 2.58585858585859 2.62626262626263 2.66666666666667 2.70707070707071 2.74747474747475 2.78787878787879 2.82828282828283 2.86868686868687 2.90909090909091 2.94949494949495 2.98989898989899 3.03030303030303 3.07070707070707 3.11111111111111 3.15151515151515 3.19191919191919 3.23232323232323 3.27272727272727 3.31313131313131 3.35353535353535 3.39393939393939 3.43434343434343 3.47474747474747 3.51515151515152 3.55555555555556 3.5959595959596 3.63636363636364 3.67676767676768 3.71717171717172 3.75757575757576 3.7979797979798 3.83838383838384 3.87878787878788 3.91919191919192 3.95959595959596 4 4 3.95918367346939 3.91836734693878 3.87755102040816 3.83673469387755 3.79591836734694 3.75510204081633 3.71428571428571 3.6734693877551 3.63265306122449 3.59183673469388 3.55102040816327 3.51020408163265 3.46938775510204 3.42857142857143 3.38775510204082 3.3469387755102 3.30612244897959 3.26530612244898 3.22448979591837 3.18367346938776 3.14285714285714 3.10204081632653 3.06122448979592 3.02040816326531 2.97959183673469 2.93877551020408 2.89795918367347 2.85714285714286 2.81632653061224 2.77551020408163 2.73469387755102 2.69387755102041 2.6530612244898 2.61224489795918 2.57142857142857 2.53061224489796 2.48979591836735 2.44897959183673 2.40816326530612 2.36734693877551 2.3265306122449 2.28571428571429 2.24489795918367 2.20408163265306 2.16326530612245 2.12244897959184 2.08163265306122 2.04081632653061 2 1.95918367346939 1.91836734693878 1.87755102040816 1.83673469387755 1.79591836734694 1.75510204081633 1.71428571428571 1.6734693877551 1.63265306122449 1.59183673469388 1.55102040816327 1.51020408163265 1.46938775510204 1.42857142857143 1.38775510204082 1.3469387755102 1.30612244897959 1.26530612244898 1.22448979591837 1.18367346938776 1.14285714285714 1.10204081632653 1.06122448979592 1.02040816326531 0.979591836734694 0.938775510204082 0.897959183673469 0.857142857142857 0.816326530612245 0.775510204081633 0.73469387755102 0.693877551020408 0.653061224489796 0.612244897959183 0.571428571428572 0.530612244897959 0.489795918367347 0.448979591836735 0.408163265306122 0.36734693877551 0.326530612244898 0.285714285714286 0.244897959183673 0.204081632653061 0.163265306122449 0.122448979591837 0.0816326530612246 0.0408163265306123 0

//0 0.0606060606060606 0.121212121212121 0.181818181818182 0.242424242424242 0.303030303030303 0.363636363636364 0.424242424242424 0.484848484848485 0.545454545454545 0.606060606060606 0.666666666666667 0.727272727272727 0.787878787878788 0.848484848484849 0.909090909090909 0.96969696969697 1.03030303030303 1.09090909090909 1.15151515151515 1.21212121212121 1.27272727272727 1.33333333333333 1.39393939393939 1.45454545454545 1.51515151515152 1.57575757575758 1.63636363636364 1.6969696969697 1.75757575757576 1.81818181818182 1.87878787878788 1.93939393939394 2 2.06060606060606 2.12121212121212 2.18181818181818 2.24242424242424 2.3030303030303 2.36363636363636 2.42424242424242 2.48484848484848 2.54545454545455 2.60606060606061 2.66666666666667 2.72727272727273 2.78787878787879 2.84848484848485 2.90909090909091 2.96969696969697 3.03030303030303 3.09090909090909 3.15151515151515 3.21212121212121 3.27272727272727 3.33333333333333 3.39393939393939 3.45454545454545 3.51515151515152 3.57575757575758 3.63636363636364 3.6969696969697 3.75757575757576 3.81818181818182 3.87878787878788 3.93939393939394 4 4.06060606060606 4.12121212121212 4.18181818181818 4.24242424242424 4.3030303030303 4.36363636363636 4.42424242424242 4.48484848484848 4.54545454545455 4.60606060606061 4.66666666666667 4.72727272727273 4.78787878787879 4.84848484848485 4.90909090909091 4.96969696969697 5.03030303030303 5.09090909090909 5.15151515151515 5.21212121212121 5.27272727272727 5.33333333333333 5.39393939393939 5.45454545454545 5.51515151515152 5.57575757575758 5.63636363636364 5.6969696969697 5.75757575757576 5.81818181818182 5.87878787878788 5.93939393939394 6 5.93877551020408 5.87755102040816 5.81632653061224 5.75510204081633 5.69387755102041 5.63265306122449 5.57142857142857 5.51020408163265 5.44897959183673 5.38775510204082 5.3265306122449 5.26530612244898 5.20408163265306 5.14285714285714 5.08163265306122 5.02040816326531 4.95918367346939 4.89795918367347 4.83673469387755 4.77551020408163 4.71428571428571 4.6530612244898 4.59183673469388 4.53061224489796 4.46938775510204 4.40816326530612 4.3469387755102 4.28571428571429 4.22448979591837 4.16326530612245 4.10204081632653 4.04081632653061 3.97959183673469 3.91836734693878 3.85714285714286 3.79591836734694 3.73469387755102 3.6734693877551 3.61224489795918 3.55102040816327 3.48979591836735 3.42857142857143 3.36734693877551 3.30612244897959 3.24489795918367 3.18367346938776 3.12244897959184 3.06122448979592 3 2.93877551020408 2.87755102040816 2.81632653061224 2.75510204081633 2.69387755102041 2.63265306122449 2.57142857142857 2.51020408163265 2.44897959183673 2.38775510204082 2.3265306122449 2.26530612244898 2.20408163265306 2.14285714285714 2.08163265306122 2.02040816326531 1.95918367346939 1.89795918367347 1.83673469387755 1.77551020408163 1.71428571428571 1.6530612244898 1.59183673469388 1.53061224489796 1.46938775510204 1.40816326530612 1.3469387755102 1.28571428571429 1.22448979591837 1.16326530612245 1.10204081632653 1.04081632653061 0.979591836734694 0.918367346938775 0.857142857142857 0.795918367346939 0.73469387755102 0.673469387755102 0.612244897959184 0.551020408163265 0.489795918367347 0.428571428571429 0.36734693877551 0.306122448979592 0.244897959183674 0.183673469387755 0.122448979591836 0.0612244897959187 0

// reset: 0 -0.0303030303030303 -0.0606060606060606 -0.0909090909090909 -0.121212121212121 -0.151515151515152 -0.181818181818182 -0.212121212121212 -0.242424242424242 -0.272727272727273 -0.303030303030303 -0.333333333333333 -0.363636363636364 -0.393939393939394 -0.424242424242424 -0.454545454545455 -0.484848484848485 -0.515151515151515 -0.545454545454545 -0.575757575757576 -0.606060606060606 -0.636363636363636 -0.666666666666667 -0.696969696969697 -0.727272727272727 -0.757575757575758 -0.787878787878788 -0.818181818181818 -0.848484848484849 -0.878787878787879 -0.909090909090909 -0.939393939393939 -0.96969696969697 -1 -1.03030303030303 -1.06060606060606 -1.09090909090909 -1.12121212121212 -1.15151515151515 -1.18181818181818 -1.21212121212121 -1.24242424242424 -1.27272727272727 -1.3030303030303 -1.33333333333333 -1.36363636363636 -1.39393939393939 -1.42424242424242 -1.45454545454545 -1.48484848484848 -1.51515151515152 -1.54545454545455 -1.57575757575758 -1.60606060606061 -1.63636363636364 -1.66666666666667 -1.6969696969697 -1.72727272727273 -1.75757575757576 -1.78787878787879 -1.81818181818182 -1.84848484848485 -1.87878787878788 -1.90909090909091 -1.93939393939394 -1.96969696969697 -2 -2.03030303030303 -2.06060606060606 -2.09090909090909 -2.12121212121212 -2.15151515151515 -2.18181818181818 -2.21212121212121 -2.24242424242424 -2.27272727272727 -2.3030303030303 -2.33333333333333 -2.36363636363636 -2.39393939393939 -2.42424242424242 -2.45454545454545 -2.48484848484848 -2.51515151515152 -2.54545454545455 -2.57575757575758 -2.60606060606061 -2.63636363636364 -2.66666666666667 -2.6969696969697 -2.72727272727273 -2.75757575757576 -2.78787878787879 -2.81818181818182 -2.84848484848485 -2.87878787878788 -2.90909090909091 -2.93939393939394 -2.96969696969697 -3									// [V] applied bias to right contact

t_switch = 1e-6, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3 , 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3		/ [s] timescale for switching		/ [s] timescale for switching

Icc = 500 														// [uA] enforced compliance current

Rs = 1e-16 														// [Ohm] series resistance 

log_freq = 10													// frequency of device snapshot generation (per x kmc steps)

output_freq = 1 												// frequency of printing the steps into the output file

verbose = 0							    						// this doesn't do anything yet

num_atoms_first_layer = 144										// number of contact atoms in the first slice
num_layers_contact = 6											// number of slices in the contact
num_atoms_contact = 864											// total number of contact atoms
num_atoms_reservoir = 540										// number of atoms in the oxygen reservoir
metals = Ti N													// these atoms will be treated as conductive

// Physical Constants
double kB = 8.617333262e-5;        								// [eV/K]
double q = 1.60217663e-19;         								// [C]
double k = 8.987552e9 / epsilon;   								// [N m^2 / C^2]
double h_bar_sq = 4.3957e-67;      								// [(Js)^2]
double h_bar = 1.054571817e-34;    								// [Js]
double h_bar_eV = 6.582119569e-16; 								// [eV]
double m_0 = 9.11e-31;             								// [kg]
double eV_to_J = 1.6e-19;          								// [C]
//double m_e = m_r * m_0;            							// [kg]

// Device constants
t_ox = 52.6838e-10                       						// [m] thickness oxide
A = 26.914773122e-10 * 26.6371955996e-10 						// [m^2] device area 
c_p = 1.92;                               						// [J/Kcm^3]
k_th_metal = 29                                					// [W/mK]
k_th_non_vacancy = 0.5                         					// [W/mK]
k_th_vacancies = 5                             					// [W/mK]

// Power update
alpha = 0.1 													// power dissipation constant

// Global thermal model
dissipation_constant = 1e-13 									// in [J/Ks] Tunable parameter
small_step = 1e-17

// Local thermal model
delta_t = 1e-13                									// step time [s] (rejection free event time == delta_t)
delta = 1                    									// [a.u.]
power_adjustment_term = 1 										// [a.u.]
L_char = 3.5e-10               									// characteristic length [m]


parameters_Pt_NiO_Pt.txt
//******************************
//  Simulation Parameters
//******************************

rnd_seed = 1							// random number for KMCProcess
restart = 0							// restart calculation
restart_xyz_file = structure_files_Pt_NiO_Pt/snapshot_init.xyz				// restart filename

//atom_xyz_file = structure_files_Pt_NiO_Pt/structure.xyz			// if not restart, draws from the atom coordinates file
//interstitial_xyz_file = structure_files/interstitials.xyz
lattice = 101.2146 29.30436 29.30436 			// 1x1 device

pristine = 0							// add initial vacancy concentration if starting from pristine
initial_vacancy_concentration = 0.05
attempt_frequency = 10e13 					// [1/s]

shift = 0							// translate cell across the periodic boundary
shifts = 0.0 0.5 0.5

nn_dist = 3.5 							// [Angstrom] neighbor distance

solve_potential = 1						// field solvers
solve_current = 1
solve_heating_global = 1
solve_heating_local = 0
								// for potential solver:
sigma = 5e-10 						// [m] gaussian broadening
epsilon = 23.0  						// [1] relative permittivity
								
								// for current solver (tunneling parameters)
m_r = 0.8 							// [1] relative effective mass
V0 = 3.0  							// [eV] defect state energy

								// for temperature solver
k_therm = 1.1         						// [W/mK] thermal conductivity
background_temp = 300 						// [K] contact temperature

pbc = 1								// toggle for periodic boundary conditions

V_switch = 15 							// [V] applied bias to right contact

t_switch = 1e-11 						// [s] timescale for switching

Icc = 3e-3 							// [A] enforced compliance current

log_freq = 50							// frequency of device snapshot generation (per x kmc steps)

output_freq = 10						// frequency of printing the steps into the output file

verbose = 0							// print all the events that are selected in the output file

num_atoms_first_layer = 97
num_atoms_contact = 980
metals = Pt

// Physical Constants
double kB = 8.617333262e-5;        				// [eV/K]
double q = 1.60217663e-19;         				// [C]
double k = 8.987552e9 / epsilon;   				// [N m^2 / C^2]
double h_bar_sq = 4.3957e-67;      				// [(Js)^2]
double h_bar = 1.054571817e-34;    				// [Js]
double h_bar_eV = 6.582119569e-16; 				// [eV]
double m_0 = 9.11e-31;             				// [kg]
double eV_to_J = 1.6e-19;          				// [C]
//double m_e = m_r * m_0;            				// [kg]

// Device constants
t_ox = 60.4838e-10                       			// [m] thickness oxide
A = 29.30436e-10 * 29.30436e-10 			// [m^2] device area 
c_p = 300;                               			// [J/Kcm^3]
k_th_metal = 29                                			// [W/mK]
k_th_non_vacancy = 0.5                         			// [W/mK]
k_th_vacancies = 5                             			// [W/mK]

// Power update
alpha = 0.1, 0.2 						// power dissipation constant [vacancy site, non-vacancy site]

// Global thermal model
dissipation_constant = 70e-5 					// in [J/Ks] Tunable parameter
small_step = 1e-16

// Local thermal model
delta_t = 1e-11                					// step time [s] (rejection free event time == delta_t)
delta = 1                    					// [a.u.]
power_adjustment_term = 0.001 					// [a.u.]
L_char = 3.5e-10               					// characteristic length [m]


src/parameters.h
//******************************
//  Global Simulation Parameters
//******************************

#include <vector>
#include <string>

// restart calculation
bool restart = 0;
std::string restart_xyz_file = "snapshot_init.xyz";

// if not restart, draws from the atom coordinates file
std::string atom_xyz_file = "structure_files/atoms.xyz";
std::string interstitial_xyz_file = "structure_files/interstitials.xyz";
// std::vector<double> lattice = {93.72586, 53.82954, 53.27438}; // 2x2 device
std::vector<double> lattice = {93.72586, 26.914773122, 26.6371955996}; // 1x1 device

// add initial vacancy concentration if starting from pristine
bool pristine = !restart;
double initial_vacancy_concentration = 0.05;
double freq = 10e13; // [1/s]
unsigned int rnd_seed = 1;

// translate cell across the periodic boundary
bool shift = !restart;
std::vector<double> shifts = {0.0, 0.5, 0.5};

// neighbor distance:
double nn_dist = 3.5; // [Angstrom]

// field solvers
bool solve_potential = 1;
bool solve_current = 1;
bool solve_heating_global = 0;
bool solve_heating_local = 1;

// for potential solver:
double G_coeff = 1;
double high_G = G_coeff * 1; //[S]
double low_G = G_coeff * 0.00000001;
double sigma = 3.5e-10; // [m]
double epsilon = 23.0;  //[1]

// for current solver (tunneling parameters):
double m_r = 0.8; // [1]
double V0 = 3.0;  // [eV]

// for temperature solver
// double alpha = 0.20; // [1]
double k_therm = 1.1;         // [W/mK]
double background_temp = 300; // [K]

// toggle for periodic boundary conditions:
bool pbc = 1;

// applied bias to right contact:
double V_switch[] = {13.00}; // [V]

// timescale for switching:
double t_switch[] = {1}; // [s]

// enforced compliance current:
double Icc = 3e-3; // 150e-6; // [A]

// frequency of device snapshot generation (per x kmc steps)
int log_freq = 10;

// frequency of printing the steps into the output file
int output_freq = 10;

// log to output.log file? (default logs to console)
bool log_to_file = 1;

// print all the events that are selected in the output file?
bool verbose = 0;

// build the device layer by layer:
int num_atoms_first_layer = 144;
int num_atoms_contact = 864;
std::vector<std::string> metals = {"Ti", "N"};

// Physical Constants
double kB = 8.617333262e-5;        // [eV/K]
double q = 1.60217663e-19;         // [C]
double k = 8.987552e9 / epsilon;   // [N m^2 / C^2]
double h_bar_sq = 4.3957e-67;      // [(Js)^2]
double h_bar = 1.054571817e-34;    // [Js]
double h_bar_eV = 6.582119569e-16; // [eV]
double m_0 = 9.11e-31;             // [kg]
double eV_to_J = 1.6e-19;          // [C]
double m_e = m_r * m_0;            // [kg]

// Device constants
const double t_ox = 52.6838e-10;                       // thickness oxide in [m]
const double A = 26.914773122e-10 * 26.6371955996e-10; // device area [m^2]
const double c_p = 1.92;                               // in [J/Kcm^3]
double k_th_metal = 29;                                // [W/mK]
double k_th_non_vacancy = 0.5;                         // [W/mK]
double k_th_vacancies = 5;                             // [W/mK]

// Power update
std::vector<double> alpha = {0.1, 0.2}; // power dissipation constant [vacancy site, non-vacancy site]

// Global thermal model
const double dissipation_constant = 70e-5; // in [J/Ks] Tunable parameter
const double small_step = 1e-16;
double event_time = 1e-13;

// Local thermal model
const double delta_t = 1e-14;                // step time [s] (rejection free event time == delta_t)
const double delta = 0.1;                    // [a.u.]
const double power_adjustment_term = 0.0001; // [a.u.]
const double L_char = 3.5e-10;               // characteristic length [m]

// Fixed parameters
const double k_th_interface = k_th_non_vacancy + (k_th_vacancies - k_th_non_vacancy) * initial_vacancy_concentration; // [W/mK]
const double tau = k_th_interface / (L_char * L_char * c_p * 1e6);                                                    // Thermal rate constant [1/s]
